lib/ts/nextjs/ssr.ts

```diff
@@ -0,0 +1,218 @@
+
+function getAuthPagePath(): string {
+    const authPagePath = SuperTokensNextjsSSRAPIWrapper.getConfigOrThrow().appInfo.websiteBasePath || "/auth";
+    return `${authPagePath}?${FORCE_LOGOUT_PATH_PARAM_NAME}=true`;
+}
+
+function getRefreshLocation(): string {
+    // The backend api routes defined in appInfo might be different from what we use here.
+    // This refresh route will be made available only by including the node handler in the next.js middleware
+    // If someone uses nextjs with SSR, and a separate backend api server, the app info data will be invalid here.
+    return "/api/auth/session/refresh";
+}
+
+async function getSSRSessionState(
+    cookies: CookiesObject | CookiesStore
+): Promise<{ state: SSRSessionState; session?: LoadedSessionContext }> {
+    const frontToken = getCookieValue(cookies, FRONT_TOKEN_NAME) || getCookieValue(cookies, FRONT_TOKEN_HEADER_NAME);
+    if (!frontToken) {
+        return { state: "front-token-not-found" };
+    }
+
+    const parsedFrontToken = parseFrontToken(frontToken);
+    if (!parsedFrontToken.isValid) {
+        return { state: "front-token-invalid" };
+    }
+
+    logDebugMessage(`Front token expires at: ${new Date(parsedFrontToken.ate)}`);
+    if (parsedFrontToken.ate < Date.now()) {
+        return { state: "front-token-expired" };
+    }
+
+    const accessToken =
+        getCookieValue(cookies, ACCESS_TOKEN_COOKIE_NAME) || getCookieValue(cookies, ACCESS_TOKEN_HEADER_NAME);
+    if (!accessToken) {
+        return { state: "access-token-not-found" };
+    }
+
+    const parsedAccessToken = await parseAccessToken(accessToken);
+    if (!parsedAccessToken.isValid) {
+        return { state: "access-token-invalid" };
+    }
+    if (!compareTokenPayloads(parsedFrontToken.payload, parsedAccessToken.payload)) {
+        return { state: "tokens-do-not-match" };
+    }
+
+    return {
+        state: "tokens-match",
+        session: {
+            userId: parsedAccessToken.payload.sub,
+            accessTokenPayload: parsedAccessToken,
+            doesSessionExist: true,
+            loading: false,
+            invalidClaims: [],
+            accessDeniedValidatorError: undefined,
+        },
+    };
+}
+
+function parseFrontToken(
+    frontToken: string
+): { payload: AccessTokenPayload["up"]; ate: number; isValid: true } | { isValid: false } {
+    try {
+        const parsedToken = JSON.parse(decodeURIComponent(escape(atob(frontToken)))) as AccessTokenPayload;
+        if (!parsedToken.uid || !parsedToken.ate || !parsedToken.up) {
+            return { isValid: false };
+        }
+        return { payload: parsedToken.up, ate: parsedToken.ate, isValid: true };
+    } catch (err) {
+        logDebugMessage(`Error while parsing front token: ${err}`);
+        return { isValid: false };
+    }
+}
+
+// TODO:
+// - Do we need to check the token version and handle ERR_JWKS_MULTIPLE_MATCHING_KEYS like in the node SDK?
```

Comment:
I don't think we do, that only comes up with really old tokens IIRC

Reply:
resolved